### ISSUE TITLE
step-63: create_downstream_cell_ordering() only works on uniformly refined meshes

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#8221]
This function in step-63 doesn't make sense except for uniform meshes:
```
template <int dim>
std::vector<unsigned int>
create_downstream_cell_ordering(const DoFHandler<dim> &dof_handler,
                                const Tensor<1, dim>   direction)
{
  std::vector<typename DoFHandler<dim>::active_cell_iterator> ordered_cells;
  ordered_cells.reserve(dof_handler.get_triangulation().n_active_cells());
  const DoFRenumbering::
    CompareDownstream<typename DoFHandler<dim>::active_cell_iterator, dim>
      comparator(direction);
  for (const auto &cell : dof_handler.active_cell_iterators())
    ordered_cells.push_back(cell);
  std::sort(ordered_cells.begin(), ordered_cells.end(), comparator);
  std::vector<unsigned int> ordered_indices;
  ordered_indices.reserve(dof_handler.get_triangulation().n_active_cells());
  for (const auto &cell : ordered_cells)
    ordered_indices.push_back(cell->index());
  return ordered_indices;
}
```

That's because the function creates an array with elements produced from `cell->index()`. But `cell->index()` does not uniquely identify cells -- it only does so within a given level. So the *level-wise* function immediately above this function does make sense, but this one does not if one has an adaptively refined mesh.

The same issue of course exists for the random shuffling function.

@tjhei, @tcclevenger: FYI.

(I have some updates to step-63, so give me time to get that approved before you work on that function.)

**Comment by tcclevenger** [comment]
Good point. We only use the active cells version for a plot in the documentation, but I imagine if someone decides to implement for adaptive (in possible extensions) this should be at least commented on.

I'm not exactly sure what a simple fix to this would be. It needs to be used as a DataOut entry. What order are the adaptive cells in for `DataVectorType::type_cell_data`? Just  `dof_handler.begin_active()` to `dof_handler.end()` order? Because this is the order `ordered_indices` needs to be in.

**Comment by tjhei** [comment]
> What order are the adaptive cells in for `DataVectorType::type_cell_data`?

The index is ``cell->active_cell_index()``. As the code doesn't run in parallel, the change should not be difficult.
